aws/src/integration/java/org/apache/iceberg/aws/lakeformation/LakeFormationTestBase.java

```diff
@@ -417,7 +431,16 @@ private static void registerResource(String s3Location) {
               .build());
       // when a resource is registered, LF will update SLR with necessary permissions which has a
       // propagation delay
-      waitForIamConsistency();
+      Awaitility.await()
+          .pollDelay(Duration.ofSeconds(1))
+          .atMost(Duration.ofSeconds(10))
+          .until(
+              () ->
+                  lakeformation
```

Comment:
I believe the same thing @nastra pointed out https://github.com/apache/iceberg/pull/8804/files#r1371698377 applies here as well.